Content.Shared/SegmentedEntity/SegmentedEntitySystem.cs

```diff
@@ -0,0 +1,306 @@
+using Robust.Shared.Physics;
+using Content.Shared.Damage;
+using Content.Shared.Explosion;
+using Content.Shared.Humanoid;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Tag;
+using Content.Shared.Storage.Components;
+using Content.Shared.Weapons.Ranged.Events;
+using Content.Shared.Teleportation.Components;
+using Robust.Shared.Map;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Physics.Components;
+using System.Numerics;
+using Robust.Shared.Network;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.SegmentedEntity
+{
+    public sealed partial class LamiaSystem : EntitySystem
+    {
+        [Dependency] private readonly TagSystem _tagSystem = default!;
+        [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+        [Dependency] private readonly SharedJointSystem _jointSystem = default!;
+        [Dependency] private readonly INetManager _net = default!;
+        [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+        Queue<(SegmentedEntitySegmentComponent segment, EntityUid lamia)> _segments = new();
+
+        [ValidatePrototypeId<TagPrototype>]
+        private const string LamiaHardsuitTag = "AllowLamiaHardsuit";
```

Comment:
But I'm honestly not sure if const will allow you to use ProtoId. It should though.
```suggestion
        private const ProtoId<TagPrototype> LamiaHardsuitTag = "AllowLamiaHardsuit";
```